pose_conversions.py

```diff
@@ -0,0 +1,262 @@
+        print(f"{roll_pitch_yaw}")
+        rpy_list.append({"convention": convention, "roll_pitch_yaw": roll_pitch_yaw})
+
+    ###
+    # Convert from Robot to Zivid (any format --> Rotation Matrix (part of Transformation Matrix))
+    ###
+    print_header("Convert from Robot to Zivid (Rotation Matrix)")
+    rotation_matrix = axis_angle_to_rotation_matrix(axis_angle)
+    print(f"Rotation Matrix from Axis Angle:\n{rotation_matrix}")
+    rotation_matrix = rotation_vector_to_rotation_matrix(rotation_vector)
+    print(f"Rotation Matrix from Rotation Vector:\n{rotation_matrix}")
+    for rotation in rpy_list:
+        rotation_matrix_from_roll_pitch_yaw = R.from_euler(
+            rotation["convention"].value, rotation["roll_pitch_yaw"]
+        ).as_matrix()
+        print(
+            f"Rotation Matrix from Roll-Pitch-Yaw angles ({rotation['convention'].name})"
+        )
+        print(f" to :\n{rotation_matrix_from_roll_pitch_yaw}")
+
+    # Replace rotation matrix in transformation matrix
+    transformation_matrix[:3, :3] = rotation_matrix
+    # Save transformation matrix which has passed through quaternion representation
+    save_transformation_matrix_to_yaml(transformation_matrix, "robotTransformOut.yaml")
+
+
+class RotationConvention(enum.Enum):
+    """Convenience enum class to list rotation conventions for Roll Pitch Yaw."""
+
+    ZYX_Intrinsic = "ZYX"
+    XYZ_Extrinsic = "xyz"
+    XYZ_Intrinsic = "XYZ"
+    ZYX_Extrinsic = "zyx"
+
+
+class AxisAngle:
+    """Convenience class to access rotation axis and angle."""
+
+    def __init__(self, axis=np.array([0, 0, 1]), angle=None):
+        """Initialize class and its variables.
+
+        Can be initialized with a unit vector and an angle, or only a rotation vector.
+
+        Args:
+            axis: rotation axis
+            angle: rotation angle
+
+        Raises:
+            ValueError: if angle vector is provided, but vector is not a unit vector
+
+        """
+        self.angle = angle
+        self.axis = axis
+        if angle is None:
+            self.angle = np.linalg.norm(axis)
+            self.axis = axis / self.angle
+        elif np.linalg.norm(axis) != 0:
+            raise ValueError(f"Angle provided, but vector is not unit vector")
+
+    def as_rotvec(self):
+        """Return rotation vector from axis angle.
+
+        Returns:
+            rotation vector
+
+        """
+        return self.axis * self.angle
+
+    def as_quaternion(self):
+        """Return quaternion from axis angle.
+
+        Returns:
+            quaternion
+
+        """
+        return R.from_rotvec(self.as_rotvec()).as_quat()
+
+
+@dataclass
+class Representations:
+    """Class to hold various transformation representations."""
+
+    axis_angle: AxisAngle() = AxisAngle()
+    rotation_vector: np.array = np.zeros(3)
+    quaternion: np.array = np.zeros(4)
+    rotations: list = field(default_factory=list)
+
+
+def zivid_to_axis_angle(transformation_matrix):
```

Comment:
I would name these type of functions as transformation_matrix_to_something as it is not consistend that in one direction you have zivid_to_something but not in the other direction i.e. something_to_zivid